vertx-opentracing/src/main/java/io/vertx/tracing/opentracing/OpenTracingTracer.java

```diff
@@ -48,88 +43,94 @@ public OpenTracingTracer(boolean closeTracer, Tracer tracer) {
-    if (sc != null) {
-      builder = builder.asChildOf(sc);
+    try (CapturedContext ignored = VertxContextScopeManager.withContext(context)) {
+      Tracer.SpanBuilder builder = tracer.buildSpan(operation);
+      if (sc != null) {
+        builder = builder.asChildOf(sc);
+      }
+      Span span = builder
+        .ignoreActiveSpan()
+        .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_SERVER)
+        .withTag(Tags.COMPONENT.getKey(), "vertx").start();
+      tracer.activateSpan(span);
+      reportTags(span, request, tagExtractor);
+      return span;
     }
-    builder
-      .ignoreActiveSpan()
-      .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_SERVER)
-      .withTag(Tags.COMPONENT.getKey(), "vertx");
-    Scope scope = builder.startActive(false);
-    reportTags(scope.span(), request, tagExtractor);
-    scope.close();
-    context.putLocal(ACTIVE_SCOPE, scope);
-    return scope;
   }
 
   @Override
-  public <R> void sendResponse(Context context, R response, Scope scope, Throwable failure, TagExtractor<R> tagExtractor) {
-    if (scope != null) {
-      context.removeLocal(ACTIVE_SCOPE);
-      Span span = scope.span();
-      if (failure != null) {
-        reportFailure(span, failure);
+  public <R> void sendResponse(Context context, R response, Span span, Throwable failure,
+    TagExtractor<R> tagExtractor) {
+    try (CapturedContext ignored = VertxContextScopeManager.withContext(context)) {
+      if (span != null) {
+        if (failure != null) {
+          reportFailure(span, failure);
+        }
+        reportTags(span, response, tagExtractor);
+        span.finish();
       }
-      reportTags(span, response, tagExtractor);
-      span.finish();
     }
   }
 
   @Override
-  public <R> Scope sendRequest(Context context, R request, String operation, BiConsumer<String, String> headers, TagExtractor<R> tagExtractor) {
-    Scope scope = context.getLocal(ACTIVE_SCOPE);
-    if (scope != null) {
-      Tracer.SpanBuilder builder = tracer.buildSpan(operation)
-        .asChildOf(scope.span())
-        .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CLIENT)
-        .withTag(Tags.COMPONENT.getKey(), "vertx");
-      Scope child = builder
-        .startActive(false);
-      reportTags(child.span(), request, tagExtractor);
-      if (headers != null) {
-        tracer.inject(child.span().context(), Format.Builtin.HTTP_HEADERS, new TextMap() {
-          @Override
-          public Iterator<Map.Entry<String, String>> iterator() {
-            throw new UnsupportedOperationException();
-          }
-          @Override
-          public void put(String key, String value) {
-            headers.accept(key, value);
-          }
-        });
+  public <R> Span sendRequest(Context context, R request, String operation,
+    BiConsumer<String, String> headers, TagExtractor<R> tagExtractor) {
+    try (CapturedContext ignored = VertxContextScopeManager.withContext(context)) {
```

Comment:
See my other comment, you should perhaps store the span in vertx context and get it from there as opposed to `activeSpan()`